src/main/java/net/hypercubemc/iris_installer/Installer.java

```diff
@@ -353,12 +356,48 @@ public void start() {
 
         frame.getContentPane().add(topPanel, BorderLayout.NORTH);
         frame.getContentPane().add(bottomPanel, BorderLayout.SOUTH);
+        
+        try {
+            Json versionInfo = Json.read(new URL("https://raw.githubusercontent.com/IrisShaders/Iris-Installer-Maven/master/installer-versions.json"));
+            if (versionInfo.asJsonMap().get("lastBreakingId").asInteger() > BREAKING_VERSION_NUMBER) {
+                invalidVersionError(frame, "Outdated Installer", "You are running an outdated Iris Installer version that can no longer install Iris.", false);
+                System.exit(0);
+            }
+        } catch (Exception e) {
+            System.err.println("Failed to check for updates!");
+            e.printStackTrace();
+            invalidVersionError(frame, "Update check failed", "Unable to check for installer updates. The installation may fail, you should try downloading the installer again.", true);
+        }
 
         frame.setVisible(true);
 
         System.out.println("Launched!");
     }
 
+    private void invalidVersionError(JFrame frame, String title, String message, boolean allowContinue) {
+        int messageType = allowContinue ? JOptionPane.WARNING_MESSAGE : JOptionPane.ERROR_MESSAGE;
+        int result = JOptionPane.showConfirmDialog(frame, message
+                + "\nDo you want to open the Iris website to get the latest version?", title,
+                JOptionPane.OK_CANCEL_OPTION, messageType);
+        if (result == -1) { // User pressed escape or X button in the corner
+            System.exit(0);
```

Comment:
Can't we just return a boolean and check for it to return?

Reply:
I don't think this should. Since it's asking for a parameter about whether it should exit already, it makes sense for it to exit, plus IMO when pressing escape it should always exit, regardless of whether just update check failed or it shouldn't continue at all, but clicking continue anyway on update check fail currently allows to launch the installer anyway.